compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -1384,27 +1446,13 @@ impl<'a, 'tcx> InferBorrowKind<'a, 'tcx> {
             place_with_id, diag_expr_id, mode
         );
 
-        match (self.capture_clause, mode) {
-            // In non-move closures, we only care about moves
-            (hir::CaptureBy::Ref, euv::Copy) => return,
-
-            // We want to capture Copy types that read through a ref via a reborrow
-            (hir::CaptureBy::Value, euv::Copy)
-                if place_with_id.place.deref_tys().any(ty::TyS::is_ref) =>
-            {
-                return;
-            }
-
-            (hir::CaptureBy::Ref, euv::Move) | (hir::CaptureBy::Value, euv::Move | euv::Copy) => {}
+        // Copy type being used as ByValue are equivalent to ImmBorrow and don't require any
```

Comment:
I must be missing something here, but how does the `captuer_information[place_id.place]` get initialized -- I guess that happens elsewhere?